server/src/pages/shot/text-tool.js

```diff
@@ -0,0 +1,205 @@
+      padding: dragDivPadding,
+      cursor: "move"
+    };
+
+    return [
+      <div draggable="true" key="drag" onDragStart={this.onDragStart.bind(this)} style={dragDivStyles}>
+        <div id="text-input" ref={this.textInput} contentEditable="true" key="text"
+           className={`${this.state.textSize} ${this.state.colorName} text`} >
+        </div>
+      </div>
+    ];
+  }
+
+  renderToolbar() {
+    return <div className="editor-header default-color-scheme"><div className="annotation-tools">
+      <Localized id="annotationTextSize">
+        <select className={`text-select`} title="Text Size" onChange={this.onChangeTextSize.bind(this)} value={this.state.textSize}>
+          <Localized id="textSizeSmall"><option value="small-text">Small</option></Localized>
+          <Localized id="textSizeMedium"><option value="medium-text">Medium</option></Localized>
+          <Localized id="textSizeLarge"><option value="large-text">Large</option></Localized>
+        </select>
+      </Localized>
+      <ColorPicker activeTool={this.state.tool}
+        setColorCallback={this.setColor.bind(this)}
+        color={this.state.color} />
+      <span className="annotation-divider"></span>
+      <Localized id="annotationTextConfirmButton">
+        <button className={`button transparent confirm-text`} id="confirm-text" onClick={this.onClickConfirm.bind(this)} title="Confirm Text">Confirm</button>
+      </Localized>
+      <Localized id="annotationTextCancelButton">
+        <button className={`button transparent cancel-text`} id="cancel-text" onClick={this.onClickCancel.bind(this)} title="Cancel Text">Cancel</button>
+      </Localized>
+    </div></div>;
+  }
+
+  setColor(color, colorName) {
+    this.setState({color, colorName});
+    this.textInput.current.focus();
+  }
+
+  onClickConfirm(e) {
+    const styles = window.getComputedStyle(this.textInput.current);
+
+    const BKG_COLOR = styles["background-color"];
+    const FONT_COLOR = styles.color;
+    const FONT_SIZE = parseInt(styles["font-size"], 10);
+    const x = this.state.left + FONT_SIZE * TEXT_INPUT_PADDING_LEFT;
+    const y = this.state.top + TEXT_INPUT_PADDING_TOP;
+
+
+    const textCanvas = document.createElement("canvas");
+    textCanvas.width = this.props.baseCanvas.width;
+    textCanvas.height = this.props.baseCanvas.height;
+    const drawingContext = textCanvas.getContext("2d");
+
+    drawingContext.scale(this.props.canvasPixelRatio, this.props.canvasPixelRatio);
+    drawingContext.textBaseline = "top";
+    drawingContext.fillStyle = `${BKG_COLOR}`;
+    drawingContext.fillRect(this.state.left,
+                            this.state.top,
+                            this.textInput.current.clientWidth,
+                            this.textInput.current.clientHeight);
+    drawingContext.fillStyle = `${FONT_COLOR}`;
+    drawingContext.font = `${FONT_WEIGHT} ${FONT_SIZE}px ${FONT_STYLE}`;
+    drawingContext.fillText(this.textInput.current.textContent, x, y);
```

Comment:
Looks like the vertical shift on confirm is from the difference between the height and line-height of the text input div.

Reply:
Seeing this subtle shift when text is rendered on Canvas for large and small text size, could slight difference in y position of fillText be causing this?

Reply:
Fixed by explicitly handling vertical shift for respective fonts